db/migrate/20180314090401_move_column_data_to_authorizations_table.rb

```diff
@@ -0,0 +1,10 @@
+class MoveColumnDataToAuthorizationsTable < ActiveRecord::Migration[5.1]
+  def change
+    User.where.not(category: [nil, '']) do |user|
+      user.authorizations.create(
+        provider: user.provider,
+        uid: user.uid
+      )
+    end
+  end
+end
```

Comment:
@sinpai why didn't you do everything in one migration?
Look:
```ruby
class RemoveProviderAndUidFromUsers < ActiveRecord::Migration[5.1]
  class User < ApplicationRecord
    has_many :authorizations
  end

  class Authorization < ApplicationRecord
    belongs_to :user
  end

  def up
    migrate_auth_data_to_authorizations

    remove_column :users, :provider
    remove_column :users, :uid
  end

  def down
    add_column :users, :provider, :string
    add_column :users, :uid, :string
  end

  private

  def migrate_auth_data_to_authorizations
    User.where.not(provider: nil).find_each do |user|
      user.authorizations.create(
        provider: user.provider,
        uid: user.uid
      )
    end
  end
end
```

------------------------------------------------------------------------------------------------

And step by step:
**1.** We've used `User` model, so we have to initialize it in migration. Why? 
```
  All our team members get this new code and run the migration. It works as expected - perfect! 
But tomorrow we decide to rename the `User` model to `Customer`. We create new migration to rename the table, 
rename the `User` model to `Customer` and don’t touch the old migrations. Tests are working, new migration is
working and everyone is happy. But what will be with this migration in one week when we run it on the production
server? It will be broken because at those moment we won’t have User model yet. We forgot to rename the User
model in the old migrations. It happens too often in Rails development. So don’t be like a kamikaze and don’t
write code to update data in migrations. Never!
```
So:
```ruby
class User < ApplicationRecord
end
```
**2.** We've used `user. authorizations` but we didn't specify it in re-initialized `User` model, so:
```ruby
class User < ApplicationRecord
  has_many :authorizations
end

class Authorization < ApplicationRecord
  belongs_to :user
end
```
**3.** Use `find_each` for safe getting of batches of entries. As a default, it gets 1000 entries each time. If you have 100_000 users -> you will probably overload data migration because all entries will be fetched at one moment
https://apidock.com/rails/ActiveRecord/Batches/ClassMethods/find_each

Reply:
I've applied this, thank you very much! 👍